File: Academy.Backend/src/CourseManagement/Academy.CourseManagement.Application/DTOs/ModuleDto.cs
namespace Academy.CourseManagement.Application.DTOs
{
    public class ModuleDto 
    {
        public Guid Id { get; set; }
        public Guid CourseId { get; set; }
        public string Title { get; set; } = string.Empty;
        public int Position { get; set; }
        public List<LessonDto> Lessons { get; set; } = new();
    }
}

File: Academy.Frontend/src/modules/courses/models/moduleDto.ts
export interface ModuleDto {
  id: string;
  courseId: string;
  title: string;
  position: number;
  lessons: LessonInfoDto[];
}

export interface LessonInfoDto {
  id: string;
  title: string;
  position: number;
  lessonType: string;
}
